### PR TITLE
Add documentation for preemption

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Follow up to #514 

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes-sigs/kueue/issues/576

#### Special notes for your reviewer:

I'm using v1beta1 in this PR, but updates to the rest of the documentation will come in a separate PR.

### CODE REVIEW

**Comment by tenzen-y** [comment]
Fixes #576

File: config/components/crd/bases/kueue.x-k8s.io_clusterqueues.yaml [hunk 1]
```diff
@@ -509,27 +509,27 @@ spec:
...
-                      cohort that are using more than their min quota. Possible values
-                      are: \n - `Never` (default): do not preempt workloads in the
-                      cohort. - `LowerPriority`: if the pending workload fits within
-                      the min quota of its ClusterQueue, only preempt workloads in
-                      the cohort that have lower priority than the pending Workload.
-                      - `Any`: if the pending workload fits within the min quota of
-                      its ClusterQueue, preempt any workload in the cohort, irrespective
+                      cohort that are using more than their nominal quota. Possible
+                      values are: \n - `Never` (default): do not preempt workloads
+                      in the cohort. - `LowerPriority`: if the pending workload fits
```
Comment (cortespao) [inline config/components/crd/bases/kueue.x-k8s.io_clusterqueues.yaml:528]:
```suggestion
                      in the cohort. - `LowerPriority`: if the pending Workload fits
```

File: site/content/en/docs/concepts/cluster_queue.md [hunk 2]
```diff
@@ -256,6 +256,55 @@ you can set the `.spec.resources[*].flavors[*].quota.max`
...
+    withinClusterQueue: LowerPriority
+```
+
+The fields above have the following semantics:
+
+- `reclaimWithinCohort` determines whether a pending Workload can preempt
+  Workloads from other ClusterQueues in the cohort that are using more than
+  their nominal quota. Possible values are:
+
+	  - `Never` (default): do not preempt workloads in the cohort.
```
Comment (cortespao) [inline site/content/en/docs/concepts/cluster_queue.md:285]: Not sure all this indentation space is needed

File: site/content/en/docs/concepts/cluster_queue.md [hunk 3]
```diff
@@ -256,6 +256,55 @@ you can set the `.spec.resources[*].flavors[*].quota.max`
...
+  preemption:
+    reclaimWithinCohort: Any
+    withinClusterQueue: LowerPriority
+```
+
+The fields above have the following semantics:
+
+- `reclaimWithinCohort` determines whether a pending Workload can preempt
+  Workloads from other ClusterQueues in the cohort that are using more than
+  their nominal quota. Possible values are:
```
Comment (cortespao) [inline site/content/en/docs/concepts/cluster_queue.md:283]:
```suggestion
  Workloads from other ClusterQueues that are using more than
  their nominal quota in the cohort. The possible values are:
```

File: site/content/en/docs/concepts/cluster_queue.md [hunk 4]
```diff
@@ -256,6 +256,55 @@ you can set the `.spec.resources[*].flavors[*].quota.max`
...
+
+	  - `Never` (default): do not preempt workloads in the cohort.
+	  - `LowerPriority`: if the pending workload fits within the nominal
```
Comment (cortespao) [inline site/content/en/docs/concepts/cluster_queue.md:286]:
```suggestion
	  - `LowerPriority`: if the pending Workload fits within the nominal
```

File: site/content/en/docs/concepts/cluster_queue.md [hunk 5]
```diff
@@ -256,6 +256,55 @@ you can set the `.spec.resources[*].flavors[*].quota.max`
...
+	    quota of its ClusterQueue, only preempt workloads in the cohort that have
+	    lower priority than the pending Workload.
+	  - `Any`: if the pending workload fits within the nominal quota of its
```
Comment (cortespao) [inline site/content/en/docs/concepts/cluster_queue.md:289]:
```suggestion
	  - `Any`: if the pending Workload fits within the nominal quota of its
```

File: site/content/en/docs/concepts/cluster_queue.md [hunk 6]
```diff
@@ -256,6 +256,55 @@ you can set the `.spec.resources[*].flavors[*].quota.max`
...
+  within the nominal quota for its ClusterQueue, can preempt active Workloads in
+  the ClusterQueue. Possible values are:
+ 
+  - `Never` (default): do not preempt workloads in the ClusterQueue.
+  - `LowerPriority`: only preempt workloads in the ClusterQueue that have
+    lower priority than the pending Workload.
+
+Note that an incoming Workload can preempt Workloads both within the
+ClusterQueue and the cohort. Kueue implements heuristics to preempt as few
+Workloads as possible, prefering Workloads with these characteristics:
```
Comment (cortespao) [inline site/content/en/docs/concepts/cluster_queue.md:303]:
```suggestion
Workloads as possible, prioritizing Workloads with these characteristics:
```

If you decide to keep  _preferring_, fix the typo :)

**Review by cortespao — COMMENTED** [review]
Just adding some suggestions.

File: site/content/en/docs/concepts/cluster_queue.md (hunk 2 above)
Comment (alculquicondor) [reply site/content/en/docs/concepts/cluster_queue.md]: removed

File: site/content/en/docs/concepts/cluster_queue.md (hunk 4 above)
Comment (alculquicondor) [reply site/content/en/docs/concepts/cluster_queue.md]: Done

File: site/content/en/docs/concepts/cluster_queue.md (hunk 5 above)
Comment (alculquicondor) [reply site/content/en/docs/concepts/cluster_queue.md]: Done

File: site/content/en/docs/concepts/cluster_queue.md (hunk 6 above)
Comment (alculquicondor) [reply site/content/en/docs/concepts/cluster_queue.md]: :joy: thanks
I kept the word "preferring". I think "prioritizing" can be confusing given that we talk about priority below.

**Review by alculquicondor — COMMENTED** [review]
/label tide/merge-method-squash

**Comment by kerthcet** [comment]
/lgtm
/hold
In case of other advices.

**Pull request merged**